src/plugins/audioplayers/plugin.c

```diff
@@ -0,0 +1,194 @@
+#include "plugins/audioplayers.h"
+
+#include <collection.h>
+#include <flutter-pi.h>
+#include <platformchannel.h>
+#include <pluginregistry.h>
+
+FILE_DESCR("audioplayers plugin")
+
+#define AUDIOPLAYERS_LOCAL_CHANNEL "xyz.luan/audioplayers"
+#define AUDIOPLAYERS_GLOBAL_CHANNEL "xyz.luan/audioplayers.global"
+
+static struct audio_player *audioplayers_linux_plugin_get_player(char *player_id, char *mode);
+
+static struct plugin {
+    struct flutterpi *flutterpi;
+    bool initialized;
+    struct concurrent_pointer_set players;
+} plugin;
+
+static int on_local_method_call(char *channel, struct platch_obj *object, FlutterPlatformMessageResponseHandle *responsehandle) {
+    const char *method;
+    int result = 1;
+
+    (void) responsehandle;
+    (void) channel;
+
+    method = object->method;
+    struct std_value *fl_player_id = stdmap_get_str(&object->std_arg, "player_id");
+    if (fl_player_id == NULL) {
+        LOG_ERROR("Call missing mandatory parameter player_id.");
+        return platch_respond_illegal_arg_std(responsehandle, "Expected `arg['player_id'] to be a string.");
+    }
+    char *player_id = STDVALUE_AS_STRING(*fl_player_id);
+
+    struct std_value *args = &object->std_arg;
+    struct std_value *fl_mode = stdmap_get_str(args, "mode");
+
+    char *mode = fl_mode == NULL ? "" : STDVALUE_AS_STRING(*fl_mode);
+
+    // CONTINE
+    struct audio_player *player = audioplayers_linux_plugin_get_player(player_id, mode);
```

Comment:
```suggestion
```suggestion
static int on_local_method_call(char *channel, struct platch_obj *object, FlutterPlatformMessageResponseHandle *responsehandle) {
    struct audio_player *player;
    struct std_value *args, *tmp;
    const char *method;
    char *player_id, *mode;
    int result = 1;

    (void) responsehandle;
    (void) channel;
    method = object->method;
    args = &object->std_arg;
    
    if (args == NULL || !STDVALUE_IS_MAP(*args)) {
        return respond_illegal_arg_std(responsehandle, "Expected `arg` to be a map.");
    }
    
    tmp = stdmap_get_str(&object->std_arg, "player_id");
    if (tmp == NULL || !STDVALUE_IS_STRING(tmp)) {
        LOG_ERROR("Call missing mandatory parameter player_id.\n");
        return platch_respond_illegal_arg_std(responsehandle, "Expected `arg['player_id'] to be a string.");
    }
    player_id = STDVALUE_AS_STRING(*tmp);

    tmp = stdmap_get_str(args, "mode");
    if (tmp == NULL) {
        mode = "";
    } else if (STDVALUE_IS_STRING(*tmp)) {
        mode = STDVALUE_AS_STRING(*tmp);
    } else {
        return platch_respond_illegal_arg_std(responsehandle, "Expected `arg['mode']` to be a string or null.");
    }
    
    player = audioplayers_linux_plugin_get_player(player_id, mode);
```
```